test/integration/utils/mock_collector_utils.go

```diff
@@ -0,0 +1,396 @@
+			}
+
+			return values
+		}
+	}
+}
+
+func WaitUntilNextScrapeCycle(t *testing.T, ctx context.Context) {
+	t.Helper()
+
+	waitCtx, cancel := context.WithTimeout(ctx, timeoutTime)
+	defer cancel()
+
+	ticker := time.NewTicker(tickerTime)
+	defer ticker.Stop()
+
+	prevScrapeValue, err := getRequestCountMetric(t, ctx)
+	if err != nil {
+		t.Fatalf("Failed to get initial scrape value: %v", err)
+		return
+	}
+
+	for {
+		select {
+		case <-waitCtx.Done():
+			t.Fatalf("Timed out waiting for new scrape cycle")
+			return
+		case <-ticker.C:
+			currentMetric, err2 := getRequestCountMetric(t, ctx)
+			if err2 != nil {
+				continue
+			}
+
+			if currentMetric != prevScrapeValue {
+				t.Log("Successfully detected new scrape cycle")
+
+				return
+			}
+		}
+	}
+}
+
+func WaitForMetricsToExist(t *testing.T, ctx context.Context) {
+	t.Helper()
+
+	waitCtx, cancel := context.WithTimeout(ctx, timeoutTime)
+	defer cancel()
+
+	ticker := time.NewTicker(tickerTime)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-waitCtx.Done():
+			t.Fatal("Timed out waiting for NGINX metrics to exist")
+			return
+		case <-ticker.C:
+			family := ScrapeCollectorMetricFamilies(t, ctx, MockCollectorStack.Otel)["nginx_http_request_count"]
+			if family != nil {
+				t.Log("NGINX metrics found")
+				return
+			}
+			t.Log("NGINX metrics not found, retrying...")
+		}
+	}
+}
+
+func getRequestCountMetric(t *testing.T, ctx context.Context) (float64, error) {
+	t.Helper()
+
+	family := ScrapeCollectorMetricFamilies(t, ctx, MockCollectorStack.Otel)["nginx_http_request_count"]
+	if family == nil {
+		return 0, fmt.Errorf("metric nginx_http_request_count not found: %v", family)
+	}
+
+	return SumMetricFamily(family), nil
+}
+
+func SumMetricFamily(metricFamily *dto.MetricFamily) float64 {
+	var total float64
+	for _, metric := range metricFamily.GetMetric() {
+		if value := metricValue(metricFamily, metric); value != nil {
+			total += *value
+		}
+	}
+
+	return total
+}
+
+func SumMetricFamilyLabel(metricFamily *dto.MetricFamily, key, val string) float64 {
+	var total float64
+	for _, metric := range metricFamily.GetMetric() {
+		labels := make(map[string]string)
+		for _, labelPair := range metric.GetLabel() {
+			labels[labelPair.GetName()] = labelPair.GetValue()
+		}
+		if labels[key] != val {
+			continue
+		}
+		if value := metricValue(metricFamily, metric); value != nil {
+			total += *value
+		}
+	}
+
+	return total
+}
+
+func metricValue(metricFamily *dto.MetricFamily, metric *dto.Metric) *float64 {
+	switch metricFamily.GetType() {
+	case dto.MetricType_COUNTER:
+		return getCounterValue(metric)
+	case dto.MetricType_GAUGE:
+		return getGaugeValue(metric)
+	case dto.MetricType_SUMMARY:
+		return getSummaryValue(metric)
+	case dto.MetricType_UNTYPED:
+		return getUntypedValue(metric)
+	case dto.MetricType_HISTOGRAM, dto.MetricType_GAUGE_HISTOGRAM:
+		return getHistogramValue(metric)
+	}
+
+	return nil
+}
+
+func getCounterValue(metric *dto.Metric) *float64 {
```

Comment:
```suggestion
func counterValue(metric *dto.Metric) *float64 {
```
Context: Golang convention is to not have Get in a function name same with GetCounter, GetGauge etc